### DISCUSSION TITLE
When a shot is fired and the trajectory of a shot

### DISCUSSION THREAD

**Discussion by cfpperche** [saul/demofile#286] [Questions]
How to detect that a shot was fired (thinking in 2d which would be x0, y0 and x1, y1) and by which player? I know there is a weapon_fire event, but is there another way? Like when an entity is created?

**Comment by saul** [comment]
I don't think it's possible to determine where a missed shot ended. If a shot hits a player you can use the `player_hurt` event to determine the trajectory of the shot by looking at the victim's and attacker's current positions. With the `weapon_fire` event you can get the player's position and angle - would that suffice?
